### ISSUE TITLE
Update Gloo ci regression tests to use oldest and newest supported versions in matrix

### ISSUE THREAD

**Issue by kdorosh** [solo-io/gloo#5358]
**Is your feature request related to a problem? Please describe.**
Today our k8s regression tests only run against one kube and one helm version.

It would be nice if we could use both the newest and oldest version we wanted to support for a release in ci. Could also be a useful tool to identify breaks in tools underneath us when we want to upgrade

**Describe the solution you'd like**
Similar to xds relay GH action matrix usage, another matrix entry for k8s / helm version

**Describe alternatives you've considered**

**Additional context**
e.g. would have caught https://github.com/helm/helm/issues/10154 sooner

**Comment by chrisgaun** [comment]
Agree. Believe we can get this done as next major add to our testing suite after scaling. Slating it for v1.11

**Comment by nfuden** [comment]
Current thought is to do this in a nightly format like we are doing elsewhere.
